### DISCUSSION TITLE
Questions about v2

### DISCUSSION THREAD

**Discussion by ziadsarour** [leancodepl/patrol#1339] [Q&A] [answered]
First of all, thank you for this awesome testing framework. It's a must have and IMHO it should be part of flutter native testing framework (especially for the native part).

I've seen that you are working on a v2 but I've not much informations about it, so here are some questions : 

- any breaking changes ?
- do you have an estimated release date ?
- will the native part be improved and enhanced ? If yes, could we have some exemple of what is in the scope ?
- in my case, I can't see any print when I run tests on iOS with Patrol, is this a known issue and if yes are you planning to solve it ?

Thank again for what you've achieved for the Flutter community !

**Comment by bartekpacia** [comment, marked as answer]
Hi, thanks for kind words :) And yes, we're going to publish v2 soon.

> any breaking changes ?

Unfortunately yes - that's why it'll be v2 (we follow semver). But they'll not be huge – you'll have to replace a few boilerplate files in `android` or `ios` folders, but Dart test code (in `integration_test`) will mostly stay the same. There'll be a few new minor limitations (see [this comment](https://github.com/leancodepl/patrol/issues/1004#issuecomment-1562852165)), but we'll likely resolve them in version 2.x.

We'll also have a migration guide.

> do you have an estimated release date ?

June or July.

> will the native part be improved and enhanced ? If yes, could we have some exemple of what is in the scope ?

The main new feature of v2 is [test bundling](https://github.com/leancodepl/patrol/issues/1004). There's lots to it, but TL;DR - it's gonna be real nice.

> in my case, I can't see any print when I run tests on iOS with Patrol, is this a known issue and if yes are you planning to solve it ?

I suspect it's one of the known issues related to logging (most likely #860). In general, you should use `patrol develop` while developing tests, and `patrol test` to run them.

---

I hope I answered your concerns :) Thanks for asking and expressing interest – really appreciate it!

**Reply by ziadsarour** [reply]
Thank you to for the answer.
It seems really great, can't wait to try it 👍
